src/mono/System.Private.CoreLib/src/System/Reflection/Metadata/AssemblyExtensions.cs

```diff
@@ -1,11 +1,69 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Runtime.CompilerServices;
+
 namespace System.Reflection.Metadata
 {
     public static class AssemblyExtensions
     {
         [CLSCompliant(false)]
         public static unsafe bool TryGetRawMetadata(this Assembly assembly, out byte* blob, out int length) => throw new NotImplementedException();
+
+        /// <summary>
+        /// Updates the specified assembly using the provided metadata, IL and PDB deltas.
+        /// </summary>
+        /// <remarks>
+        /// Currently executing methods will continue to use the existing IL. New executions of modified methods will
+        /// use the new IL. Different runtimes may have different limitations on what kinds of changes are supported,
+        /// and runtimes make no guarantees as to the state of the assembly and process if the delta includes
+        /// unsupported changes.
+        /// </remarks>
+        /// <param name="assembly">The assembly to update.</param>
+        /// <param name="metadataDelta">The metadata changes to be applied.</param>
+        /// <param name="ilDelta">The IL changes to be applied.</param>
+        /// <param name="pdbDelta">The PDB changes to be applied.</param>
+        /// <exception cref="ArgumentNullException">The assembly argument is null.</exception>
+        /// <exception cref="NotSupportedException">The update could not be applied.</exception>
+        public static void ApplyUpdate(Assembly assembly, ReadOnlySpan<byte> metadataDelta, ReadOnlySpan<byte> ilDelta, ReadOnlySpan<byte> pdbDelta = default)
```

Comment:
Once https://github.com/dotnet/runtime/pull/48366 is in, can we refactor this to have most of this method in the shared partition?  Seems like the main difference is just the name of the ApplyUpdate_internal helper; we should be able to name the helpers the same and have just those pieces in the individual runtime partitions.

Reply:
Getting some conflicting guidance here https://github.com/dotnet/runtime/pull/48380#discussion_r578846729 :smile: 

But actually maybe your suggestion was to just leave the QCall / InternallCall runtime-specific and to share the invocations.  Unfortunately that won't work today - the arguments are a bit different for each runtime.

Reply:
> maybe your suggestion was to just leave the QCall / InternallCall runtime-specific and to share the invocations

Yeah, that's what I was hoping for.  If it's not possible, ok.